src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,185 @@
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            addIfFirst(value);
+        } else {
+            addIfLast(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (index == size) {
+            addIfLast(value);
+        } else {
+            Node<T> standStill = searchNode(index).prev;
+            Node<T> newNode = new Node<>(standStill, value, searchNode(index));
+            searchNode(index).prev = newNode;
+            if (standStill == null) {
+                first = newNode;
+            } else {
+                standStill.next = newNode;
+            }
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        T[] inputArray = (T[]) list.toArray();
+        int lengthInputArray = inputArray.length;
+        if (lengthInputArray == 0) {
+            return;
+        }
+        Node<T> standStill = last;
+        for (T element : inputArray) {
+            Node<T> newNode = new Node<>(standStill, element, null);
+            if (standStill == null) {
+                first = newNode;
+            } else {
+                standStill.next = newNode;
+            }
+            standStill = newNode;
+        }
+        last = standStill;
+        size += lengthInputArray;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexForGetAndSet(index);
+        return searchNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexForGetAndSet(index);
+        Node<T> foundNode = searchNode(index);
+        T oldValue = foundNode.item;
+        foundNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexForGetAndSet(index);
+        return unlink(searchNode(index));
     }
 
     @Override
-    public boolean remove(T object) {
+    public boolean remove(T value) {
+        if (value == null) {
+            for (Node<T> node = first; node != null; node = node.next) {
+                if (node.item == null) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> node = first; node != null; node = node.next) {
+                if (value.equals(node.item)) {
+                    unlink(node);
+                    return true;
+                }
+            }
```

Comment:
You have two same loops. The only difference is the `if` statement. Can we use only one loop and one `if` in this method?